### ISSUE TITLE
pnpm link -g, happens an error: has no binaries

### ISSUE THREAD

**Issue by dribble-njr** [pnpm/pnpm#7511]
### Verify latest release

- [X] I verified that the issue exists in the latest pnpm release

### pnpm version

v8.14.1

### Which area(s) of pnpm are affected? (leave empty if unsure)

CLI

### Reproduction steps

`pnpm link -g`

xxx has no binaries

node: 20.10.0
pnpm: 8.14.1
npm: 10.2.3

### Describe the Bug

run `pnpm link -g`

xxx has no binaries

### Expected Behavior

pnpm link -g success

### Which Node.js version are you using?

20.10.0

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by zkochan** [comment]
What is the bug? It doesn't fail, it prints a warning.

Linking package globally is usually useful if they are used as CLI tools.

**Comment by dribble-njr** [comment]
My `package.json`, and my `cli.js` has the right node shebang.

```
{
  "bin": {
      "tp":` "./bin/cli.js"
  }
}
```

Then I run `pnpm link --global` and run `tp`. It can't work.

If I changed it to `npm link`, it worked.

**Comment by zkochan** [comment]
Maybe try to add a name field to your package.json

**Comment by await-ovo** [comment]
> Then I run `pnpm link --global` and run `tp`. It can't work.
> 
> If I changed it to `npm link`, it worked.

I also tested it on my side and `tp` seems to be working fine, but the warning probably shouldn't be there ~

**Comment by zkochan** [comment]
This must be a duplicate of https://github.com/pnpm/pnpm/issues/4761

**Comment by wilfreud** [comment]
Checked the docs: https://pnpm.io/cli/unlink
Used `npm uninstall --global <package>` as specified. 

Plus I wanted to change the name of the binary, so since `pnpm unlink` doesn't do the job, I had to go to `C:\Users\<username>\AppData\Local\pnpm` and remove the files with my bin field  name. Franly at this point it's 1AM and I don't know what I am doing. I did that used pnpm unlink too. somehow worked for me

**Comment by conradkoh** [comment]
For anyone else who has the issue, I had to run `pnpm setup` to get the binary to link. There were 2 issues for me
1. The warning was printing
2. The command was not executing regardless of what I tried

**Troubleshooting**
1. Run the command `pnpm bin -g` - this should show something
2. If it is empty, run pnpm setup

Once I got the bin folder, the warning was still showing, but the command was working.
